### PR TITLE
Feat(web-react): Make the preferred height and max height of `ModalDialog` customizable #DS-723

### PR DESCRIPTION
**Pull request by adamkudrna**
New API options:

- `maxHeightFromTabletUp`
- `preferredHeightOnMobile`
- `preferredHeightFromTabletUp`

image
image

### CODE REVIEW

File: packages/web-react/src/components/ModalComposed/ModalComposedDialog.tsx [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
+    preferredHeightOnMobile,
+    preferredHeightFromTabletUp,
+    ...restProps
+  } = props;
 
   const { classProps } = useModalComposedStyleProps({ isExpandedOnMobile });
   const { styleProps, props: otherProps } = useStyleProps(restProps);
 
+  const myStyle: CSSPropertiesWithVariables = {
+    color: 'red', // TODO delete once everything is working
```
Comment (literat) [inline packages/web-react/src/components/ModalComposed/ModalComposedDialog.tsx:31]: I am just marking this to do not forget to remove it.

File: packages/web-react/src/components/ModalComposed/ModalComposedDialog.tsx [hunk 2]
```diff
@@ -1,18 +1,44 @@
...
+    maxHeightFromTabletUp,
+    preferredHeightOnMobile,
+    preferredHeightFromTabletUp,
+    ...restProps
+  } = props;
 
   const { classProps } = useModalComposedStyleProps({ isExpandedOnMobile });
   const { styleProps, props: otherProps } = useStyleProps(restProps);
 
+  const myStyle: CSSPropertiesWithVariables = {
```
Comment (literat) [inline packages/web-react/src/components/ModalComposed/ModalComposedDialog.tsx:30]: Can we do better naming? For example `customizedHeightStyle`...

File: packages/web-react/src/components/ModalComposed/ModalComposedDialog.tsx [hunk 3]
```diff
@@ -1,18 +1,44 @@
-import React, { ElementType, ForwardedRef, forwardRef } from 'react';
+import React, { CSSProperties, ElementType, ForwardedRef, forwardRef } from 'react';
 import classNames from 'classnames';
 import { useStyleProps } from '../../hooks';
 import { ModalComposedDialogProps, ModalComposedDialogElementType } from '../../types/modalComposed';
 import { useModalComposedStyleProps } from './useModalComposedStyleProps';
 
+interface CSSPropertiesWithVariables extends CSSProperties {
```
Comment (literat) [inline packages/web-react/src/components/ModalComposed/ModalComposedDialog.tsx:7]: In this case, I am also suggesting a more descriptive name, `CustomizedHeightCSSProperties`... `CSSPropertiesWithVariables` is too general for this case, IMHO. We are dealing with height customization, so I would use that word in the naming. :-)

**Comment by adamkudrna** [comment]
@literat I'll need help with ESlint, I'm afraid…

image

Works on my machine, of course.

**Comment by literat** [comment]
> @literat I'll need help with ESlint, I'm afraid…
> 
> image
> 
> Works on my machine, of course.

Works on my machine, too. 🤔

**Comment by literat** [comment]
> > @literat I'll need help with ESlint, I'm afraid…
> > image
> > Works on my machine, of course.
> 
> Works on my machine, too. 🤔

Solved, I think. See https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1

```diff
+ /* eslint-disable import/no-extraneous-dependencies, import/extensions, import/no-unresolved */
- // eslint-disable import/no-extraneous-dependencies, import/extensions, import/no-unresolved
```

**Review by literat — APPROVED** [review]
Nice :-) I have solved the issue with ESLint so this is ready to be merged.

**Pull request merged**